app/controllers/ally_controller.rb

```diff
@@ -0,0 +1,142 @@
+class AllyController < ApplicationController
+
+  #
+  # REST methods
+  #
+
+  ##
+  # Returns whether the plugin is enabled for the account. If its
+  # enabled extra information such as the client ID and base URL
+  # of the Ally API is returned as well
+  def enabled
+    plugin_data = get_plugin_data
+    data = {
+      :enabled => plugin_data[:enabled],
+      :clientId => plugin_data[:client_id],
+      :baseUrl => plugin_data[:base_url]
+    }
+    render :json => data
+  end
+
+  ##
+  # Sign a request to the Ally REST API but don't execute it.
+  # The UI will execute the request
+  def sign
+    if (request = get_request())
+      plugin_data = get_plugin_data()
+      puts request.to_yaml
+      data = {
+        :clientId => plugin_data[:client_id],
+        :baseUrl => plugin_data[:base_url],
+        :path => request.path,
+        :header => request.to_hash["authorization"][0],
+        :body => request.body,
+      }
+      render :json => data
+    end
+  end
+
+  ##
+  # Proxy data from the Ally REST API
+  def proxy
+    if (request = get_request())
+      # TODO: Cache the HTTP object per hostname allowing
+      # for connection pooling
+      plugin_data = get_plugin_data()
+      uri = URI.parse(plugin_data[:base_url])
+      http = Net::HTTP.new(uri.host, uri.port)
+      response = http.request(request)
+
+      # Render successful requests as JSON
+      if response.code == 200 || response.code == 201
+        render :json => response.body, :status => response.code
+      else
+        render :text => response.body, :status => response.code
+      end
```

Comment:
When I try `body: response.body` it complains it can't find a view. I could try having an empty view, but that seems kinda weird. There is a `Rack::Proxy` object that I'll look into in a bit, maybe that helps